src/main.py

```diff
@@ -59,11 +59,14 @@ def render_template(template, *args, **kwargs):
     language = get_language(lang)
     langcode_length = len(lang) + 1 # Probably always 2-character language codes but who knows!
 
-    # If the template does not exist, then redirect to English version
-    if (lang != '' and lang != 'en' and not(os.path.isfile('templates/%s' % template))):
-        return redirect('/en%s' % (request.full_path[langcode_length:]), code=302)
-
-    # Although a langauge may be enabled, all templates may not have been translated yet
+    # If the template does not exist, then redirect to English version if it exists, else home
+    if (lang != '' and not(os.path.isfile('templates/%s' % template))):
```

Comment:
Unrelated thought. I wonder if it's necessary to call `isfile` each time as opposed to memoizing results, since we wouldn't expect the result to change.

Reply:
Remember that we cache all pages for 3 hours per CDN so, in effect, they are memorised for a bit for subsequent users in that location. Plus it's still pretty fast.

We could generate it as part of `npm run generate` of similar, and this goes back to you Static Site suggestion in #634. And similar to what I said back then, I don't think the performance hit is actually that noticeable, and the migration would be a large effort. So for this PR I say leave it as is.